src/script/ceph-backport.sh

```diff
@@ -1619,10 +1620,11 @@ fi
 
 milestone_number=$(try_known_milestones "$milestone")
 if [ "$milestone_number" -gt "0" ] >/dev/null 2>&1 ; then
-    target_branch="$milestone"
+    true
 else
     milestone_number=$(milestone_number_from_remote_api "$milestone")
 fi
```

Comment:
This kinda looks like the conditional should be rewritten. I'm not sure what this code does but seems like this can end up comparing `[ "TBD" -gt "0" ]`. Using a string with arithmetic comparison might make for interesting bugs down the road.
In any case this could also be
```
if [ "$milestone_number" -le "0" ] >/dev/null 2>&1 ; then
    milestone_number=$(milestone_number_from_remote_api "$milestone")
fi
```

Reply:
Very perceptive! And indeed "TBD" -gt "0" is exactly what it does when the base branch is not known to the script, and that triggers the else condition.

```
$ test "TBD" -gt "0"
bash: test: TBD: integer expression expected
$ echo $?
2
```

You can think of this as:

```
if [ is_int "$milestone_number" ] ; then
    true
else
    milestone_number=$(milestone_number_from_remote_api "$milestone")
fi
```

(This rarely-used code path was not working, so the script was failing for octopus backports - this patch fixes it. I'll change the "true" to a debug message to make the `if` block less offensive.)

Reply:
ahh ok I see.  Maybe a comment that `-eg` is abused as a `is_int()` here might be good?